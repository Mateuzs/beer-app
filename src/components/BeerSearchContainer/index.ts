export { default as BeerSearchContainer } from "./BeerSearchContainer";
