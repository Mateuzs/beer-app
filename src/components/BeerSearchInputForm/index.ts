export { default as BeerSearchInputForm } from "./BeerSearchInputForm";
