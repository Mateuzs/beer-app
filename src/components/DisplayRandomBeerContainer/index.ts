export { default as DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
