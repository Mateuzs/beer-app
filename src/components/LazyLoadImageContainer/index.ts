export { default as LazyLoadImageContainer } from "./LazyLoadImageContainer";
