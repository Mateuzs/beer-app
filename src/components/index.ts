export { BeerSearchContainer } from "./BeerSearchContainer";
export { DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
export { NavBar } from "./NavBar";
export { SearchButton } from "./SearchButton";
export { LazyLoadImageContainer } from "./LazyLoadImageContainer";
export { ErrorMessage } from "./ErrorMessage";
export { BeerSearchInputForm } from "./BeerSearchInputForm";
export { BeerList } from "./BeerList";
