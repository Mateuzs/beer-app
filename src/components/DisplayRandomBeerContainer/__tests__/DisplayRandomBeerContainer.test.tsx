// external
import React from "react";
import { shallow, mount } from "enzyme";
// components
import { DisplayRandomBeerContainer } from "../../../components";

const mockedValidResponse = [
  {
    id: 0,
    name: "BEER",
    image_url: "IMAGE_URL",
    description: "DESCRIPTION",
  },
];

jest.mock("axios", () => ({
  get: (url: string) => {
    switch (url) {
      case "https://api.punkapi.com/v2/beers/random":
        return {
          data: mockedValidResponse,
        };
    }
  },
}));

describe("DisplayRandomBeerContainer component", () => {
  test("should match the snapshot and styles", () => {
    const mockedData = {
      isFetching: false,
      setIsFetching: jest.fn(),
      setIsError: jest.fn(),
    };
    const wrapper = shallow(<DisplayRandomBeerContainer {...mockedData} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should render expected text", async () => {
    const mockedData = {
      isFetching: false,
      setIsFetching: jest.fn(),
      setIsError: jest.fn(),
    };
    const component = mount(<DisplayRandomBeerContainer {...mockedData} />);

    expect(component.find('[data-test-id="random-beer-description"]').text()).toEqual("Loading...");
    expect(component.find('[data-test-id="random-beer-title"]').text()).toEqual("Loading...");
    expect(component.find('[data-test-id="random-beer-image"]').length).toEqual(1);
    expect(component.find('[data-test-id="random-beer-description-buttons"]').length).toEqual(1);

    await new Promise((r) => setTimeout(r, 1000));

    expect(component.find('[data-test-id="random-beer-description"]').text()).toEqual(
      "DESCRIPTION"
    );
    expect(component.find('[data-test-id="random-beer-title"]').text()).toEqual("BEER");
  });
});
