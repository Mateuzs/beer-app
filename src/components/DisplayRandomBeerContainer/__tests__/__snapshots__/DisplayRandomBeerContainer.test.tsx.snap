// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayRandomBeerContainer component should match the snapshot and styles 1`] = `
<div
  className="random-beer-container"
>
  <p
    className="random-beer-title"
    data-test-id="random-beer-title"
  >
    Loading...
  </p>
  <div
    className="random-beer-description-container"
  >
    <div
      className="random-beer-image"
      data-test-id="random-beer-image"
    >
      <Memo(LazyLoadImageContainer)
        height={300}
        imageUrl="Loading..."
      />
    </div>
    <div
      className="random-beer-description"
      data-test-id="random-beer-description"
    >
      Loading...
    </div>
    <div
      className="random-beer-description-buttons"
      data-test-id="random-beer-description-buttons"
    >
      <Memo(SearchButton)
        isDisabled={false}
        label="Another beer!"
        onClickHandler={[Function]}
      />
      <Memo(SearchButton)
        isDisabled={false}
        label="Random non alcoholic beer"
        onClickHandler={[Function]}
      />
    </div>
  </div>
</div>
`;
